ui/components/app/collectibles-tab/collectibles-tab.test.js

```diff
@@ -0,0 +1,308 @@
+      address: '0x806627172af48bd5b0765d3449a7def80d6576ff',
+      config: '',
+    },
+    image:
+      'https://lh3.googleusercontent.com/4jfPi-nQNWCUXD5qVNVWX7LX2UufU_elEJcvICFlsTdcBXv70asnDEOlI8oKECZxlXq1wseeIXMwmP5tLyOUxMKk',
+    standard: 'ERC1155',
+  },
+  {
+    address: '0x495f947276749Ce646f68AC8c248420045cb7b5e',
+    tokenId:
+      '58076532811975507823669075598676816378162417803895263482849101571116612190209',
+    name: 'Punk #4651',
+    creator: {
+      user: {
+        username: null,
+      },
+      profile_img_url: null,
+      address: '0x806627172af48bd5b0765d3449a7def80d6576ff',
+      config: '',
+    },
+    image:
+      'https://lh3.googleusercontent.com/BdxvLseXcfl57BiuQcQYdJ64v-aI8din7WPk0Pgo3qQFhAUH-B6i-dCqqc_mCkRIzULmwzwecnohLhrcH8A9mpWIZqA7ygc52Sr81hE',
+    standard: 'ERC1155',
+  },
+  {
+    address: '0xDc7382Eb0Bc9C352A4CbA23c909bDA01e0206414',
+    tokenId: '1',
+    name: 'MUNK #1',
+    description: null,
+    image: 'ipfs://QmTSZUNt8AKyDabkyXXXP4oHWDnaVXgNdXoJGEyaYzLbeL',
+    standard: 'ERC721',
+  },
+  {
+    address: '0xDc7382Eb0Bc9C352A4CbA23c909bDA01e0206414',
+    tokenId: '2',
+    name: 'MUNK #2',
+    description: null,
+    image: 'ipfs://QmTSZUNt8AKyDabkyXXXP4oHWDnaVXgNdXoJGEyaYzLbeL',
+    standard: 'ERC721',
+  },
+  {
+    address: '0xDc7382Eb0Bc9C352A4CbA23c909bDA01e0206414',
+    tokenId: '3',
+    name: 'MUNK #3',
+    description: null,
+    image: 'ipfs://QmTSZUNt8AKyDabkyXXXP4oHWDnaVXgNdXoJGEyaYzLbeL',
+    standard: 'ERC721',
+  },
+];
+
+const COLLECTIBLES_CONTRACTS = [
+  {
+    address: '0x495f947276749Ce646f68AC8c248420045cb7b5e',
+    name: 'PUNKS',
+    symbol: 'PNKS',
+    schemaName: 'ERC1155',
+  },
+  {
+    address: '0xDc7382Eb0Bc9C352A4CbA23c909bDA01e0206414',
+    name: 'Munks',
+    symbol: 'MNKS',
+  },
+];
+
+const ACCOUNT_1 = '0x123';
+const ACCOUNT_2 = '0x456';
+
+const render = ({
+  collectibleContracts = [],
+  collectibles = [],
+  selectedAddress,
+  chainId = '0x1',
+  collectiblesDetectionNoticeDismissed = false,
+  useCollectibleDetection,
+  onAddNFT = jest.fn(),
+}) => {
+  const store = configureStore({
+    metamask: {
+      allCollectibles: {
+        [ACCOUNT_1]: {
+          [chainId]: collectibles,
+        },
+      },
+      allCollectibleContracts: {
+        [ACCOUNT_1]: {
+          [chainId]: collectibleContracts,
+        },
+      },
+      provider: { chainId },
+      selectedAddress,
+      collectiblesDetectionNoticeDismissed,
+      useCollectibleDetection,
+    },
+  });
+  return renderWithProvider(<CollectiblesTab onAddNFT={onAddNFT} />, store);
+};
+
+describe('Collectible Items', () => {
+  const detectCollectiblesStub = jest.fn();
+  const setCollectiblesDetectionNoticeDismissedStub = jest.fn();
+  setBackgroundConnection({
+    setCollectiblesDetectionNoticeDismissed: setCollectiblesDetectionNoticeDismissedStub,
+    detectCollectibles: detectCollectiblesStub,
+  });
+
+  const useHistoryOriginal = reactRouterDom.useHistory;
```

Comment:
[nit] What do you think about using `jest.mock` to stub `useHistory`? Then you shouldn't have to worry about clearing mocks or returning to the original implementation.

Reply:
This actually wasn't required at all. Removed here: https://github.com/MetaMask/metamask-extension/pull/12970/commits/624c55749d7151df368e3d6ecdd5b50675deda70